driver/src/price_finding/naive_solver.rs

```diff
@@ -497,12 +472,110 @@ pub mod tests {
                 buy_amount: 19960,
             },
         ];
+        let state = create_account_state_with_balance_for(&orders);
 
-        let mut solver = NaiveSolver::new(Some(Fee {
+        let fee = Some(Fee {
             token: 2,
             percentage: 0.001,
-        }));
+        });
+        let mut solver = NaiveSolver::new(fee.clone());
         let res = solver.find_prices(&orders, &state).unwrap();
         assert_eq!(res.executed_sell_amounts, [0, 0]);
+
+        check_solution(&orders, res, &fee).unwrap();
+    }
+
+    fn order_pair_first_fully_matching_second() -> Vec<Order> {
+        vec![
+            Order {
+                batch_information: None,
+                account_id: H160::from(1),
+                sell_token: 0,
+                buy_token: 1,
+                sell_amount: 52 * 10u128.pow(18),
+                buy_amount: 4 * 10u128.pow(18),
+            },
+            Order {
+                batch_information: None,
+                account_id: H160::from(0),
+                sell_token: 1,
+                buy_token: 0,
+                sell_amount: 15 * 10u128.pow(18),
+                buy_amount: 180 * 10u128.pow(18),
+            },
+        ]
+    }
+
+    fn order_pair_both_fully_matched() -> Vec<Order> {
+        vec![
+            Order {
+                batch_information: None,
+                account_id: H160::from(1),
+                sell_token: 2,
+                buy_token: 1,
+                sell_amount: 10 * 10u128.pow(18),
```

Comment:
Why the need to use 10^18 here?

Reply:
They are still valid. However in reality we are always going to be trading amounts in the area of 10**18 and thus in order to get more confidence that our e2e tests would also work I increased the numbers to be in the same order of magnitude.

Reply:
However, knowing that this is all working for smaller instances is still relevant for unit testing.